### ISSUE TITLE
connection_string for Azure Service Bus will always be replaced in commercetools_subscription

### ISSUE THREAD

**Issue by dsanchenz** [labd/terraform-provider-commercetools#320]
Terraform version: 1.1.5
Provider version: 1.5.1
Resource: commercetools_subscription

On every terraform apply,

Destroy and add the commercetools_subscription resource because it finds changes to the connection_string and forces it to be replaced.

In provider version 0.31.0 it was not replaced

```hcl
resource "commercetools_subscription" "ct_orders_notification_commercetools_subscription" {
  key = "ct-orders-notification"
  destination {
    type              = "AzureServiceBus"
    connection_string = "${data.terraform_remote_state.terraform_azure.outputs.environment_servicebus_namespace_primary_connection_string};EntityPath=${data.terraform_remote_state.terraform_azure.outputs.ct_orders_notification_servicebus_topic_name}"
  }
  message {
    resource_type_id = "order"
    types            = ["OrderStateTransition"]
  }
}
```

**Comment by therealppa** [comment]
I thank this might be because when the connection string is queried from ctp with GET it will be partially encoded with ***** for security reasons - so it will not match the value that's in the terraform file.

https://docs.commercetools.com/api/projects/subscriptions#azureservicebusdestination

**Comment by therealppa** [comment]
Just in case someone needs it, the issue can be ignored by using

  lifecycle {
    ignore_changes = [
      # TODO: Watch https://github.com/labd/terraform-provider-commercetools/issues/320 for possible resolution
      destination["connection_string"]
    ]
  }

Of course this means if the connection string _does_ change you'll have to force the resource replacement manually.

**Comment by mvantellingen** [comment]
Should be fixed in the next release

**Comment by mvantellingen** [comment]
Should be fixed in 1.6.0 (just released). Feel free to re-open if there are still issues
